torch/csrc/jit/script/compiler.cpp

```diff
@@ -535,6 +536,44 @@ Value* tryCreateList(
   return graph.insertNode(graph.createList(elem_type, list_ctor))->output();
 }
 
+// Match against a potentially mutable schema.
+//
+// We need to treat mutable schemas differently because the IR explicitly
+// expresses effects by including a world token in mutable ops. Users do not
+// know about the world token, so we need to generate a dummy one and add
+// it to the inputs for schema matching.
+//
+// Example:
+//   append(int[] list, int el)
+// becomes
+//   append(World w, int[] list, int el)
+//
+// NOTE: The dummy world token has no meaning; the AnnotateEffects pass is
+// necessary to enforce linearization on effectful ops.
+at::optional<std::vector<Value*>> tryMatchMutableSchema(
+    const FunctionSchema& schema,
+    const SourceRange& loc,
+    Graph& graph,
+    at::ArrayRef<NamedValue> inputs,
+    at::ArrayRef<NamedValue> attributes,
+    std::ostream& failure_messages,
+    bool convert_tensors_to_nums) {
+  std::vector<NamedValue> modifiedInputs(inputs.begin(), inputs.end());
+  if (schema.is_mutable) {
+    // Add a dummy world token to be matched against
+    const auto worldToken = graph.insertConstant(World());
```

Comment:
A `prim::Constant` world and `prim::EntryWorld` would have the same behavior the in annotate effects pass. If we just inserted a node `prim::LoadCurrentWorld` in both cases you can unify the concept.

Reply:
Just to clarify about naming: `prim::Store` takes stack values and puts them to registers, returning them as output. `prim::Load` takes the registers and puts them on the stack. 

So in this case, prim::EntryWorld is more like prim::Store, right? Since it is taking something from the outside world and spawning it on the stack.